cogs/pug.py

```diff
@@ -606,18 +693,75 @@ def format_game_server_status(self):
     #########################################################################################
     # Functions:
     #########################################################################################
+    def removeServerReference(self, serverref: str):
+        if serverref in self.current_serverrefs() and serverref not in [None, '']:
+            self.allServers.pop(self.current_serverrefs().index(serverref))
+            return True
+        return False
+
+    def updateServerReference(self, serverref: str, serverdesc: str, serverurl: str = ''):
+        if serverref in self.current_serverrefs() and serverref not in [None, '']:
+            self.allServers.pop(self.current_serverrefs().index(serverref))
+        self.allServers.append((serverref,serverdesc,serverurl))
+        return True
+    
+    def useServer(self, index: int):
+        """Sets the active server"""
+        if index >= 0 and index <= len(self.allServers):
+            self.gameServerRef = self.allServers[index-1][0]
```

Comment:
You could try to access self.allServers[-1] here. It should either be:
1-based: index >= 1 and index <= len(self.allServers)
0-based: index >= 0 and index < len(self.allServers)